editoast, front, core: remove gamma type

bougue-pe opened the pull request:
Part of https://github.com/OpenRailAssociation/osrd/issues/9708

Rename `gamma` to `const_gamma` to reflect its use: constant gamma braking coefficient used when NOT circulating under ETCS/ERTMS signaling system in m/s^2.

All rolling stock used are `CONST`, front [always provided `CONST`](https://github.com/OpenRailAssociation/osrd/pull/9887/files#diff-a720b5041d0ecae646392425aae1d15acb49a0b9749fa9fad25161de0226551aL158-L161), this `MAX` variant was never used, and actually always constant in code use (and not sure if a max-check was missing in [computeAcceleration](https://github.com/OpenRailAssociation/osrd/pull/9887/files#diff-ab3c0ecd0ebc92a826fb7aebe46ecfc25ccf08f0ecd7a4a3f33dc363ea1434b5R105))

🔍 please review by commit.

To merge right after this:
- [x] corresponding PR in dags https://github.com/osrd-project/osrd-dags/pull/596
- [x] probably corresponding PR in https://github.com/y-plus/pyOSRD/pull/113 > useless as pinned to a previous version

bougue-pe committed:
core: remove unused python script for rolling-stock migration

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
editoast, front, core: remove gamma type

rename gamma to const_gamma to reflect its use: constant gamma braking
coefficient used when NOT circulating under ETCS/ERTMS signaling system in
m/s^2.

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

Erashin reviewed (approved):
Core changes work for me :)

woshilapin reviewed (approved):
The PR seems good. However, I'm missing some understanding about why it's fine to remove some (possibly not really well-supported) functionality: why removing `MAX`? It'd be nice to at least explain in the PR description why it was there? Why we decided to remove it? Just put some tracing about this decision so "future us" understand why, at this point in time, we decided to remove `MAX`.

shenriotpro reviewed (approved):
Thanks

emersion reviewed (approved):
LGTM for `front/` changes (not tested)

bougue-pe committed:
core: remove maxBrakingForce from RS interface

All use of braking is currently constant in code, and in database.
So it is now explicit.

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe commented on core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java:
```diff
@@ -53,9 +53,6 @@ static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCur
     double getDeceleration();
 
-    /** The maximum braking force which can be applied at a given speed, in newtons */
-    double getMaxBrakingForce(double speed);
```
@woshilapin wrote:
> The PR seems good. However, I'm missing some understanding about why it's fine to remove some (possibly not really well-supported) functionality: why removing `MAX`? It'd be nice to at least explain in the PR description why it was there? Why we decided to remove it? Just put some tracing about this decision so "future us" understand why, at this point in time, we decided to remove `MAX`.

Added this commit to actually clean the code a bit more after your review.
Also added a little word on main comment of the PR.
🙏 Thanks !

woshilapin replied:
Thank you 😄

achrafmohye reviewed (approved):
LGTM 👍
